netComparator/cmd/main.go

```diff
@@ -153,62 +125,77 @@ func action(c *cli.Context) error {
 		return fmt.Errorf("failed to marshall transactions from mainnet: %v", err)
 	}
 
-	retryConfig = []retry.Option{
+	retryConfig := []retry.Option{
 		retry.OnRetry(func(n uint, err error) {
 			log.Info("Retry request", n+1, err)
 		}),
 		retry.Attempts(retries),
 		retry.Delay(5 * time.Second),
 	}
 
-	wrappedDiffs := compareTransactions(txHashes)
+	wrappedDiffs := compareTransactions(txHashes, retryConfig)
 
-	jsonBytes, err := json.MarshalIndent(wrappedDiffs, "", " ")
+	err = generateOutputReport(wrappedDiffs, config.Outfile)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to generate HTML report: %v", err)
 	}
 
-	err = os.WriteFile(config.Outfile, jsonBytes, 0644)
-	if err != nil {
-		return err
-	}
+	return nil
+}
 
-	// Execute the list in the go template.
-	tmpl, err := template.ParseFS(fs, "assets/template.html")
-	if err != nil {
-		panic(err)
+func newProxies(primaryUrl, secondaryUrl string) (wrappedProxy, wrappedProxy, error) {
+	primary := blockchain.ArgsProxy{
+		ProxyURL:            primaryUrl,
+		Client:              nil,
+		SameScState:         false,
+		ShouldBeSynced:      false,
+		FinalityCheck:       false,
+		CacheExpirationTime: time.Minute,
+		EntityType:          core.Proxy,
 	}
 
-	file, _ := os.Create("index.html")
-	defer file.Close()
+	secondary := blockchain.ArgsProxy{
+		ProxyURL:            secondaryUrl,
+		Client:              nil,
+		SameScState:         false,
+		ShouldBeSynced:      false,
+		FinalityCheck:       false,
+		CacheExpirationTime: time.Minute,
+		EntityType:          core.Proxy,
+	}
 
-	err = tmpl.Execute(file, wrappedDiffs)
+	// Create a client for the mainnet.
+	pp, err := blockchain.NewProxy(primary)
 	if err != nil {
-		return fmt.Errorf("failed to execute template: %v", err)
+		return nil, nil, fmt.Errorf("failed to create primary proxy: %v", err)
 	}
 
-	log.Info("finished creating report")
+	// Create a client for the shadow-fork.
```

Comment:
Comments are now a little bit misleading (since we mean "primary" vs. "secondary").

Also, perhaps `primaryProxy` instead of `pp` and `secondaryProxy` instead of `sp`?